### PR TITLE
Remove Jackson usage from Health Insights, Purview Sharing, and some samples and testing

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Remove Jackson usage in the following source code:

- azure-health-insights-cancerprofiling
- azure-health-insights-clinicalmatching
- azure-analytics-purview-sharing

And miscellaneous removal Jackson usage in sample and test code.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/healthinsights/azure-health-insights-cancerprofiling/src/main/java/com/azure/health/insights/cancerprofiling/AzureHealthInsightsServiceVersion.java [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
 
 import com.azure.core.util.ServiceVersion;
 
-/** Service version of CancerProfilingClient. */
-public enum CancerProfilingServiceVersion implements ServiceVersion {
-    /** Enum value 2023-03-01-preview. */
+/**
+ * Service version of AzureHealthInsightsClient.
+ */
+public enum AzureHealthInsightsServiceVersion implements ServiceVersion {
```
Comment (srnagar) [inline sdk/healthinsights/azure-health-insights-cancerprofiling/src/main/java/com/azure/health/insights/cancerprofiling/AzureHealthInsightsServiceVersion.java:12]: This type name should not be changed.

File: sdk/healthinsights/azure-health-insights-clinicalmatching/src/main/java/com/azure/health/insights/clinicalmatching/AzureHealthInsightsServiceVersion.java [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
 
 import com.azure.core.util.ServiceVersion;
 
-/** Service version of ClinicalMatchingClient. */
-public enum ClinicalMatchingServiceVersion implements ServiceVersion {
-    /** Enum value 2023-03-01-preview. */
+/**
+ * Service version of AzureHealthInsightsClient.
+ */
+public enum AzureHealthInsightsServiceVersion implements ServiceVersion {
```
Comment (srnagar) [inline sdk/healthinsights/azure-health-insights-clinicalmatching/src/main/java/com/azure/health/insights/clinicalmatching/AzureHealthInsightsServiceVersion.java:12]: Same here - this type name should not be changed.

**Review by srnagar — APPROVED** [review]

File: sdk/healthinsights/azure-health-insights-cancerprofiling/src/main/java/com/azure/health/insights/cancerprofiling/AzureHealthInsightsServiceVersion.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/healthinsights/azure-health-insights-cancerprofiling/src/main/java/com/azure/health/insights/cancerprofiling/AzureHealthInsightsServiceVersion.java:12]: This was a change in the generated code, not sure how it had the old names.

Comment (alzimmermsft) [reply sdk/healthinsights/azure-health-insights-cancerprofiling/src/main/java/com/azure/health/insights/cancerprofiling/AzureHealthInsightsServiceVersion.java:12]: Will change in a follow-up PR

File: sdk/healthinsights/azure-health-insights-clinicalmatching/src/main/java/com/azure/health/insights/clinicalmatching/AzureHealthInsightsServiceVersion.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/healthinsights/azure-health-insights-clinicalmatching/src/main/java/com/azure/health/insights/clinicalmatching/AzureHealthInsightsServiceVersion.java:12]: Will change in a follow-up PR

**Pull request merged**